File: C++版/20. 有效的括号.cpp
class Solution {
public:
    bool isValid(string s) {
        int n = s.length();
        stack<char> sta;
        for(int i = 0;i < n; i++){
            if(s[i] == '(')
                sta.push(')');
            else if(s[i] == '[')
                sta.push(']');
            else if(s[i] == '{')
                sta.push('}');
            else if(sta.empty())   //用来处理[']']这种情况
                return false;
            else if(sta.top() != s[i])
                return false;
            else
                sta.pop();
        }
        return sta.empty();
    }
};

File: Python版/20. 有效的括号.py
class Solution:
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []  #使用list来代替栈
        for i in s:
            if i == '(':
                stack.append(')')
            elif i == '{':
                stack.append('}')
            elif i == '[':
                stack.append(']')
            elif len(stack) == 0:
                return False
            elif stack[-1] != i:
                return False
            else:
                stack.pop()
        return len(stack) == 0